app/qml/LoadingIndicator.qml

```diff
@@ -0,0 +1,81 @@
+import QtQuick 2.7
+import QtQuick.Controls 2.2
+import QtGraphicalEffects 1.0
+
+ProgressBar {
+    id: loadingIndicator
+    anchors.top: parent.top
+    value: 0
+    clip: true
+
+    property int easingType: Easing.InOutQuad
+
+    function toggle(value) {
+      if (loadingIndicatorAnimation.running === value) return
+
+      loadingIndicatorAnimation.running = value
+      visible = value
+    }
```

Comment:
Can we simplify this? Instead of toggle() method, just something like "isRunning" property, which will be assigned to "visible" and "loadingIndicatorAnimation.running" ?